chia/_tests/pools/test_pool_cmdline.py

```diff
@@ -1,20 +1,1050 @@
+    mock_get.return_value.__aenter__.return_value.text.return_value = json.dumps(pool_response_dict)
+
+    with pytest.raises(CliRpcConnectionError, match="Incorrect version"):
+        await JoinPlotNFTCMD(
+            rpc_info=NeedsWalletRPC(
+                client_info=client_info,
+            ),
+            id=wallet_id,
+            pool_url="",
+            dont_prompt=True,
+        ).run()
+
+    pool_response_dict["relative_lock_height"] = 5
+    pool_response_dict["protocol_version"] = 1
+    mock_get.return_value.__aenter__.return_value.text.return_value = json.dumps(pool_response_dict)
+
+    # Join the new pool - this will leave the prior pool and join the new one
+    # Here you can use None as the wallet_id and the code will pick the only pool wallet automatically
+    await JoinPlotNFTCMD(
+        rpc_info=NeedsWalletRPC(
+            client_info=client_info,
+        ),
+        id=None,
+        pool_url="http://127.0.0.1",
+        dont_prompt=True,
+    ).run()
+
+    await wallet_environments.full_node.farm_blocks_to_puzzlehash(count=3, guarantee_transaction_blocks=True)
+    await verify_pool_state(wallet_rpc, wallet_id, PoolSingletonState.LEAVING_POOL)
+    await wallet_environments.full_node.farm_blocks_to_puzzlehash(count=12, guarantee_transaction_blocks=True)
+    await verify_pool_state(wallet_rpc, wallet_id, PoolSingletonState.FARMING_TO_POOL)
+    await wallet_environments.full_node.wait_for_wallet_synced(
+        wallet_node=wallet_environments.environments[0].node, timeout=20
+    )
+
+    # Create a second farming plotnft to url http://pool.example.com
+    wallet_id = await create_new_plotnft(wallet_environments, self_pool=False, second_nft=True)
+
+    # Join the new pool - this will leave the prior pool and join the new one
+    # Will fail because we don't specify a wallet ID and there are multiple pool wallets
+    with pytest.raises(CliRpcConnectionError, match="More than one pool wallet"):
+        await JoinPlotNFTCMD(
+            rpc_info=NeedsWalletRPC(
+                client_info=client_info,
+            ),
+            id=None,
+            pool_url="http://127.0.0.1",
+            dont_prompt=True,
+        ).run()
+
+    # Join the new pool - this will leave the prior pool and join the new one and specific wallet_id
+    await JoinPlotNFTCMD(
+        rpc_info=NeedsWalletRPC(
+            client_info=client_info,
+        ),
+        id=wallet_id,
+        pool_url="http://127.0.0.1",
+        dont_prompt=True,
+    ).run()
+
+    await wallet_environments.full_node.farm_blocks_to_puzzlehash(count=3, guarantee_transaction_blocks=True)
+    await verify_pool_state(wallet_rpc, wallet_id, PoolSingletonState.LEAVING_POOL)
+    await wallet_environments.full_node.farm_blocks_to_puzzlehash(count=12, guarantee_transaction_blocks=True)
+    await verify_pool_state(wallet_rpc, wallet_id, PoolSingletonState.FARMING_TO_POOL)
+
+    # Join the same pool test - code not ready yet for test
```

Comment:
How about now? :)

Reply:
Needs my other PR: https://github.com/Chia-Network/chia-blockchain/pull/18822
In the current code, this actually doesn't fail at all and leaves and re-joins the same pool again